cdap-security/src/main/java/io/cdap/cdap/security/server/LDAPLoginService.java

```diff
@@ -0,0 +1,94 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.server;
+
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+
+import java.util.Hashtable;
+
+import javax.naming.Context;
+import javax.naming.NamingEnumeration;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+import javax.naming.ldap.InitialLdapContext;
+import javax.naming.ldap.LdapContext;
+
+/** LDAPLoginService
+ *
+ * Class extending JAASLoginService to add support for server case conversion
+ */
+public class LDAPLoginService extends JAASLoginService {
+    private static final Logger LOG = Log.getLogger(LDAPLoginService.class);
+
+    @Override
+    public String serverCaseConvert(String userName) {
+        try {
+            Hashtable map = new Hashtable(configuration.getAppConfigurationEntry(
+                    io.cdap.cdap.security.server.LDAPLoginModule.class.getName())[0].getOptions());
+            map.put(Context.INITIAL_CONTEXT_FACTORY, map.get("contextFactory").toString());
+            map.put(Context.SECURITY_AUTHENTICATION, map.get("authenticationMethod").toString());
+            map.put(Context.PROVIDER_URL, getLDAPURL(map));
+            map.put(Context.SECURITY_PRINCIPAL, map.get("bindDn").toString());
+            map.put(Context.SECURITY_CREDENTIALS, map.get("bindPassword").toString());
+
+            if (map.containsKey("useLdaps") && map.get("useLdaps").toString().equalsIgnoreCase("true")) {
+                map.put("java.naming.ldap.factory.socket", LDAPLoginModule.TrustAllSSLSocketFactory.class.getName());
+            }
+
+            LdapContext ctx = new InitialLdapContext(map, null);
+
+            SearchControls constraints = new SearchControls();
+            constraints.setSearchScope(SearchControls.SUBTREE_SCOPE);
+
+            String ldapHandlerBinding = ctx.getEnvironment().get("userBaseDn").toString();
+
+            String filterExpr = "(" + ctx.getEnvironment().get("userRdnAttribute").toString() + "=" + userName + ")";
+
+            NamingEnumeration answer = ctx.search(ldapHandlerBinding, filterExpr, constraints);
+
+            if (answer.hasMore()) {
+                Attributes attrs = ((SearchResult) answer.next()).getAttributes();
+                userName = (String) attrs.get(ctx.getEnvironment().get("userRdnAttribute").toString()).get();
+                LOG.debug("applying case adjustment using data from LDAP, updated username:: " + userName);
+            } else {
+                LOG.warn("case adjustment not done for username, user not found in LDAP:: " + userName);
+            }
+
+        } catch (Exception e) {
+            LOG.warn("Exception occurred while get user details from LDAP, not able to apply case adjustment "
```

Comment:
Use `{}` place holder instead of `+`. Also, please log the exception.

`LOG.warn("Exception .... ", e);`

Reply:
done